contracts/cw-proposal-single/src/contract.rs

```diff
@@ -711,9 +713,90 @@ pub fn query_info(deps: Deps) -> StdResult<Binary> {
+    struct BetaProposal {
+        pub title: String,
+        pub description: String,
+        /// The address that created this proposal.
+        pub proposer: Addr,
+        /// The block height at which this proposal was created. Voting
+        /// power queries should query for voting power at this block
+        /// height.
+        pub start_height: u64,
+        /// The minimum amount of time this proposal must remain open for
+        /// voting. The proposal may not pass unless this is expired or
+        /// None.
+        pub min_voting_period: Option<Expiration>,
+        /// The the time at which this proposal will expire and close for
+        /// additional votes.
+        pub expiration: Expiration,
+        /// The threshold at which this proposal will pass.
+        pub threshold: Threshold,
+        /// The total amount of voting power at the time of this
+        /// proposal's creation.
+        pub total_power: Uint128,
+        /// The messages that will be executed should this proposal pass.
+        pub msgs: Vec<CosmosMsg<Empty>>,
+
+        pub status: Status,
+        pub votes: Votes,
+        pub allow_revoting: bool,
+
+        /// Information about the deposit that was sent as part of this
+        /// proposal. None if no deposit.
+        pub deposit_info: Option<CheckedDepositInfo>,
+    }
+
+    match msg {
+        MigrateMsg::FromBeta {} => {
+            // Retrieve current map from storage
+            let current_map: Map<u64, BetaProposal> = Map::new("proposals");
+            let count = current_map
+                .keys(deps.storage, None, None, Order::Ascending)
+                .count();
+            println!("keys count {}", count);
+
+            let current = current_map
+                .range(deps.storage, None, None, Order::Ascending)
+                .collect::<StdResult<Vec<(u64, BetaProposal)>>>()?;
+
+            // Add migrated entries to new map.
+            // TODO: determine whether this needs to be paginated
+            current
+                .into_iter()
+                .try_for_each::<_, StdResult<()>>(|(id, prop)| {
+                    let migrated_proposal = SingleChoiceProposal {
+                        title: prop.title,
+                        description: prop.description,
+                        proposer: prop.proposer,
+                        start_height: prop.start_height,
+                        min_voting_period: prop.min_voting_period,
+                        expiration: prop.expiration,
+                        threshold: prop.threshold,
+                        total_power: prop.total_power,
+                        msgs: prop.msgs,
+                        status: prop.status,
+                        votes: prop.votes,
+                        allow_revoting: prop.allow_revoting,
+                        deposit_info: prop.deposit_info,
+                        created: env.block.time,
```

Comment:
does anyone know of a better way to get the created timestamp based on the start_time block height?

Reply:
Maybe for these where we don't actually know the time we could just set it to zero. That way the frontend could tell the difference between a proposal actually created at this time, and one that was migrated.

Could also put the block height there. Wouldn't be surprised if the block height version of a timestamp was LONG before Juno was created and that way a frontend could do this calculation for us.

Reply:
I think @NoahSaso had a PR recently that dealt with this if you have any thoughts!

Reply:
wasn't able to find a way to query for block time given height in cosmwasm, so going to set it to 0 for now.